File: IS/front/src/ts/data/Position.ts
enum Position {
    DIRECTOR = 'DIRECTOR',
    COOK = "COOK",
    CLEANER = 'CLEANER',
    MANAGER_OF_CLEANING = 'MANAGER_OF_CLEANING'
}

export const positionToString = new Map<Position, string>()
positionToString.set(Position.DIRECTOR, "Управляющий")
positionToString.set(Position.COOK, "Повар")
positionToString.set(Position.CLEANER, "Уборщик")
positionToString.set(Position.MANAGER_OF_CLEANING, "Менеджер по клинингу")

export default Position

File: back/src/main/java/com/example/back/model/dto/Position.java
package com.example.back.model.dto;

public enum Position {
    DIRECTOR,
    COOK,
    CLEANER,
    MANAGER_OF_CLEANING;

    public static Position stringToPosition(String s) {
        return switch (s.toLowerCase()) {
            case "director" -> Position.DIRECTOR;
            case "cook" -> Position.COOK;
            case "cleaner" -> Position.CLEANER;
            case "manager_of_cleaning" -> Position.MANAGER_OF_CLEANING;
            default -> null;
        };
    }
}
